server/ec2spotmanager/models.py

```diff
@@ -218,7 +254,7 @@ def getMissingParameters(self):
         return missing_fields
 
 
-@receiver(models.signals.post_delete, sender=PoolConfiguration)
+@receiver(models.signals.post_delete, sender=PoolConfiguration.provider)
```

Comment:
I doubt this works and you reported that file deletion didn't work in your environment. sender here must be a class, not a class field.